dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/RumFeature.kt

```diff
@@ -236,25 +235,8 @@ internal class RumFeature(
 
         initializeVitalMonitor(CPUVitalReader(), cpuVitalMonitor, periodInMs)
         initializeVitalMonitor(MemoryVitalReader(), memoryVitalMonitor, periodInMs)
-
-        val vitalFrameCallback = VitalFrameCallback(frameRateVitalMonitor) { initialized.get() }
-        try {
-            Choreographer.getInstance().postFrameCallback(vitalFrameCallback)
-        } catch (e: IllegalStateException) {
-            // This can happen if the SDK is initialized on a Thread with no looper
-            internalLogger.log(
-                InternalLogger.Level.ERROR,
-                InternalLogger.Target.MAINTAINER,
-                "Unable to initialize the Choreographer FrameCallback",
-                e
-            )
-            internalLogger.log(
-                InternalLogger.Level.WARN,
-                InternalLogger.Target.USER,
-                "It seems you initialized the SDK on a thread without a Looper: " +
-                    "we won't be able to track your Views' refresh rate."
-            )
-        }
+        jankStatsActivityLifecycleListener = JankStatsActivityLifecycleListener(frameRateVitalMonitor, internalLogger)
+        (appContext as? Application)?.registerActivityLifecycleCallbacks(jankStatsActivityLifecycleListener)
```

Comment:
should we modify the way we are checking refresh rate? currently we are using `RumViewScope#detectRefreshRateScale`, which relies on the view key and the presence of activity/fragment there (which may be missing in case of manual instrumentation or calls coming from cross-platform).

But now, since we are listening for activities anyway, we can access UI context at any time and basically pass this info with a sample, making things work properly even for cross-platform.

wdyt?